packages/react-components/react-aria/src/hooks/useARIAButton.ts

```diff
@@ -1,22 +1,40 @@
 import { Enter, Space } from '@fluentui/keyboard-keys';
 import { resolveShorthand, useEventCallback } from '@fluentui/react-utilities';
 import type { ExtractSlotProps, ResolveShorthandFunction, Slot } from '@fluentui/react-utilities';
+import * as React from 'react';
 
-export type ARIAButtonSlotProps = ExtractSlotProps<Slot<'button', 'a'>> & {
+export type ARIAButtonType = 'button' | 'a' | 'div';
+
+export type ARIAButtonProps<Type extends ARIAButtonType = ARIAButtonType> = React.PropsWithRef<
+  JSX.IntrinsicElements[Type]
+> & {
   disabled?: boolean;
+  /**
+   * When set, allows the button to be focusable even when it has been disabled.
+   * This is used in scenarios where it is important to keep a consistent tab order
+   * for screen reader and keyboard users. The primary example of this
+   * pattern is when the disabled button is in a menu or a commandbar and is seldom used for standalone buttons.
+   *
+   * @default false
+   */
   disabledFocusable?: boolean;
 };
 
+export type ARIAButtonSlotProps = ExtractSlotProps<Slot<'button', 'a' | 'div'>> &
+  Pick<ARIAButtonProps, 'disabled' | 'disabledFocusable'>;
+
 /**
+ * @internal
+ *
  * Button keyboard handling, role, disabled and tabIndex implementation that ensures ARIA spec
  * for multiple scenarios of shorthand properties. Ensuring 1st rule of ARIA for cases
  * where no attribute addition is required.
  */
-export const useARIAButton: ResolveShorthandFunction<ARIAButtonSlotProps> = (shorthand, options) => {
-  const shorthandProps = resolveShorthand(shorthand, options);
-
-  const { disabled, disabledFocusable, onClick, onKeyDown, onKeyUp, tabIndex } = (shorthandProps ||
-    {}) as ARIAButtonSlotProps;
+export function useARIAButton<Type extends ARIAButtonType, Props extends ARIAButtonProps<Type>>(
```

Comment:
```suggestion
export function useARIAButtonProps<Type extends ARIAButtonType, Props extends ARIAButtonProps<Type>>(
```

would that make the naming more relevant ?